src/UI/Component/Input/Field/Factory.php

```diff
@@ -205,4 +206,59 @@ public function dependantGroup(array $inputs);
 	 * @return    \ILIAS\UI\Component\Input\Field\Checkbox
 	 */
 	public function checkbox($label, $byline = null);
+
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     A TagInput is used to choose a subset amount of tags out of a finite list
+	 *     of tags. The TagInput is used whenever not all available tags to choose from
+	 *     MUST or SHOULD be visible, e.g. because the amount is too high.
+	 *     Besides the tags to choose from, the user can provide own tags by typing them
+	 *     into the Input
+	 *   composition: >
+	 *     The Input is presented as a TextInput and prepended by already selected tags
+	 *     presented as texts including a Close-Button.  (e.g. [ Amsterdam X ] )
+	 *     The input is labeled by the $label given.
+	 *     Suggested tags are listed in a Dropdown-List beneath the TextInput.
+	 *   effect: >
+	 *     As soon as the user types in the TextInput, the TagInput suggests matching tags from
+	 *     the the given list of tags. Suggestions will appear after a defined
+	 *     amount of characters, one by default.
+	 *     Clicking on one of these tags closes the List and transfers the selected tag into
+	 *     the Input, displayed as a Tag with a Close-Button.
+	 *     By clicking on a Close-Button of a already selected tag, this tag will disappear
+	 *     from the Input.
+	 *
+	 * rivals: >
+	 *   SelectInput: Currently not part of the UI-Service.
+	 *
+	 * rules:
+	 *   usage:
+	 *     1: >
+	 *      A TagInput MUST NOT be used whenever a user has to perform a binary choice where
+	 *      option is automatically the inverse of the other. A Checkbox MUST be used in this case.
+	 *     2: >
+	 *      A TagInput MUST NOT be used whenever a user has to perform a choice from a list of
+	 *      options where only one Option has to be selected. A Select MUST be used in this case
+	 *      (Not yet part of the KitchenSink).
+	 *     3: >
+	 *      A TagInput MUST be used whenever a User should be able to extend the list of given options.
```

Comment:
Why is this a MUST? I can extend given options in Data collection with + options and this is quite OK. Should this be categorized as a violation of kitchensink rules after we accept this PR?

Reply:
I agree, downgrade this to SHOULD or even MAY.